packages/quick_actions/quick_actions_android/test/quick_actions_android_test.dart

```diff
@@ -4,169 +4,98 @@
-        final QuickActionsAndroid quickActions = buildQuickActionsPlugin();
-        quickActions.initialize((String type) {});
-        quickActions.clearShortcutItems();
-
-        expect(
-          log,
-          <Matcher>[
-            isMethodCall('getLaunchAction', arguments: null),
-            isMethodCall('clearShortcutItems', arguments: null),
-          ],
-        );
-      });
-
-      test('clearShortcutItems', () {
-        final QuickActionsAndroid quickActions = buildQuickActionsPlugin();
-        quickActions.clearShortcutItems();
-        expect(
-          log,
-          <Matcher>[
-            isMethodCall('clearShortcutItems', arguments: null),
-          ],
-        );
-        log.clear();
-      });
-    });
+  test('setShortCutItems', () async {
+    await quickActions.initialize((String type) {});
+    const ShortcutItem item =
+        ShortcutItem(type: 'test', localizedTitle: 'title', icon: 'icon.svg');
+    await quickActions.setShortcutItems(<ShortcutItem>[item]);
+
+    expect(api.items.first.type, item.type);
+    expect(api.items.first.localizedTitle, item.localizedTitle);
+    expect(api.items.first.icon, item.icon);
   });
 
-  group('$ShortcutItem', () {
-    test('Shortcut item can be constructed', () {
-      const String type = 'type';
-      const String localizedTitle = 'title';
-      const String icon = 'foo';
+  test('clearShortCutItems', () {
+    quickActions.initialize((String type) {});
+    const ShortcutItem item =
+        ShortcutItem(type: 'test', localizedTitle: 'title', icon: 'icon.svg');
+    quickActions.setShortcutItems(<ShortcutItem>[item]);
+    quickActions.clearShortcutItems();
+
+    expect(api.items.isEmpty, true);
+  });
 
-      const ShortcutItem item =
-          ShortcutItem(type: type, localizedTitle: localizedTitle, icon: icon);
+  test('Shortcut item can be constructed', () {
+    const String type = 'type';
+    const String localizedTitle = 'title';
+    const String icon = 'foo';
 
-      expect(item.type, type);
-      expect(item.localizedTitle, localizedTitle);
-      expect(item.icon, icon);
-    });
+    const ShortcutItem item =
+        ShortcutItem(type: type, localizedTitle: localizedTitle, icon: icon);
+
+    expect(item.type, type);
+    expect(item.localizedTitle, localizedTitle);
+    expect(item.icon, icon);
   });
 }
 
-/// This allows a value of type T or T? to be treated as a value of type T?.
-///
-/// We use this so that APIs that have become non-nullable can still be used
-/// with `!` and `?` on the stable branch.
-T? _ambiguate<T>(T? value) => value;
+class _FakeQuickActionsApi implements AndroidQuickActionsApi {
+  List<ShortcutItem> items = <ShortcutItem>[];
+  String? launchAction;
+
+  @override
+  Future<void> clearShortcutItems() {
+    items = <ShortcutItem>[];
+    return Future<void>.value();
+  }
+
+  @override
+  Future<String?> getLaunchAction() {
+    launchAction = LAUNCH_ACTION_STRING;
```

Comment:
This seems like an odd side effect; if the goal is to detect the method being called it can just be a bool tracking that (e.g., `getLaunchActionCalled`).